src/processor/model/mod.rs

```diff
@@ -0,0 +1,162 @@
+pub mod otel_transform;
+mod py_transform;
+
+use crate::processor::py::PyResourceSpans;
+use pyo3::prelude::*;
+use std::ffi::CString;
+use std::sync::{Arc, Mutex};
+use tower::BoxError;
+use tracing::error;
+
+#[derive(Debug, Clone)]
+pub struct AnyValue {
+    pub value: Arc<Mutex<Option<Value>>>,
+}
+
+#[derive(Debug, Clone)]
+#[allow(clippy::enum_variant_names)]
+pub enum Value {
+    StringValue(String),
+    BoolValue(bool),
+    IntValue(i64),
+    DoubleValue(f64),
+    ArrayValue(ArrayValue),
+    KvListValue(KeyValueList),
+    BytesValue(Vec<u8>),
+}
+
+#[derive(Debug, Clone)]
+pub struct ArrayValue {
+    pub values: Vec<AnyValue>,
+}
+
+#[derive(Debug, Clone)]
+pub struct KeyValueList {
+    pub values: Vec<opentelemetry_proto::tonic::common::v1::KeyValue>,
+}
+
+#[derive(Debug, Clone)]
+pub struct KeyValue {
+    pub key: Arc<Mutex<String>>,
+    pub value: Arc<Mutex<Option<AnyValue>>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Resource {
+    pub attributes: Arc<Mutex<Vec<Arc<Mutex<KeyValue>>>>>,
+    pub dropped_attributes_count: Arc<Mutex<u32>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct ResourceSpans {
+    pub resource: Arc<Mutex<Option<Resource>>>,
+    pub scope_spans: Arc<Mutex<Vec<Arc<Mutex<ScopeSpans>>>>>,
+    pub _schema_url: Arc<Mutex<String>>,
```

Comment:
Is there a reason for why `dropped_attributes_count`, a u32, and this schema String require an `Arc<Mutex<>>`, but the similar types in Span don't use them? Is the span type a WIP or is it because that's a leaf node in the trace model?

Reply:
Yeah, good eye. I think those can be removed if there is an Arc/Mutex on the parent struct and they are only read/written via that ArcMutex, similar to the Span as you mentioned. The ArcMutex on `dropped_attributes_count` and the scheme string are just some artifacts from early exploration. I plan on doing several more passes through this code as we become more familiar with pyo3. 

Likewise I also suspect there are some needless `clone()` calls on arcs that can be removed when accessing as struct members that are already stored in Arcs. My plan to address is comprehensive tests to verify, however a lot of the tests are currently in the .py code. That will make it difficult to produce coverage reports, so I will probably also provide pure Rust side tests so we can ascertain coverage.